core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1252,6 +1254,28 @@ private[spark] class DAGScheduler(
     execCores.map(cores => properties.setProperty(EXECUTOR_CORES_LOCAL_PROPERTY, cores))
   }
 
+  /**
+   * If push based shuffle is enabled, set the shuffle services to be used for the given
+   * shuffle map stage. The list of shuffle services is determined based on the list of
+   * active executors tracked by block manager master at the start of the stage.
```

Comment:
so I think just reading this last sentence threw me off a bit with the description you made. You said we could use all executors we have seen - those 2 things don't match because here we say active executors.

"When DRA kicks in later on, and significantly reduces the number of available active executors, we can still look into the historical locations of past executors to get sufficient shuffle service locations to perform block push/merge."

Reply:
This is my bad, earlier we didn't have support for DRA and things changed recently. Will fix this one.